File: src/main/java/com/yuyutian/mytools/wechat/account/model/WechatAccount.java
package com.yuyutian.mytools.wechat.account.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

/**
 * 微信账号实体类
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class WechatAccount {
    /** 账号ID */
    private Long id;
    /** 微信ID/微信号 */
    private String wechatId;
    /** 昵称 */
    private String nickname;
    /** 备注 */
    private String remark;
    /** 状态: 1-正常, 2-禁用 */
    private Integer status;
    /** 刷新频率: 1-每天一次, 2-每天两次, 默认2 */
    private Integer refreshFrequency;
    /** 创建时间 */
    private LocalDateTime createTime;
    /** 更新时间 */
    private LocalDateTime updateTime;
}

File: webapp/src/service/api/wechat-account.ts
import { request } from '../request';

/** 微信账号项 */
interface WechatAccount {
  id: number;
  wechatId: string;
  nickname: string;
  remark: string;
  status: number;
  createTime: string;
  updateTime: string;
}

/** 账号列表响应 */
interface AccountListResponse {
  list: WechatAccount[];
  total: number;
}

/** 获取所有正常账号 */
export function fetchGetNormalAccounts() {
  return request<WechatAccount[]>({
    url: '/api/wechat/accounts/normal',
    method: 'get'
  });
}

/** 分页获取账号列表 */
export function fetchGetAccountList(params: { page: number; pageSize: number }) {
  return request<AccountListResponse>({
    url: '/api/wechat/accounts',
    method: 'get',
    params
  });
}

/** 创建账号 */
export function fetchCreateAccount(data: {
  wechatId: string;
  nickname: string;
  remark?: string;
}) {
  return request<WechatAccount>({
    url: '/api/wechat/accounts',
    method: 'post',
    data
  });
}

/** 更新账号 */
export function fetchUpdateAccount(data: { id: number; nickname: string; remark?: string }) {
  return request({
    url: '/api/wechat/accounts',
    method: 'put',
    data
  });
}

/** 更新账号状态 */
export function fetchUpdateAccountStatus(id: number, status: number) {
  return request({
    url: `/api/wechat/accounts/${id}/status`,
    method: 'put',
    params: { status }
  });
}

/** 刷新账号下所有任务状态 */
export function fetchRefreshAccountTasks(id: number) {
  return request({
    url: `/api/wechat/accounts/${id}/refresh`,
    method: 'put'
  });
}

/** 删除账号 */
export function fetchDeleteAccount(id: number) {
  return request({
    url: `/api/wechat/accounts/${id}`,
    method: 'delete'
  });
}